jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/AppengineApiConfiguration.java

```diff
@@ -0,0 +1,138 @@
+import org.eclipse.jetty.webapp.AbstractConfiguration;
+import org.eclipse.jetty.webapp.WebAppClassLoader;
+import org.eclipse.jetty.webapp.WebAppContext;
+
+import java.util.logging.Level;
+
+public class AppengineApiConfiguration extends AbstractConfiguration {
+
+  // A class to check if the GAE API is available.
+  public static final String GAE_CHECK_CLASS = "com.google.appengine.api.ThreadManager";
+
+  // Hide the all container classes from the webapplication
+  // TODO update to use '.' when supported by Jetty
+  private static final String[] SERVER_CLASSES = {
+    "com.", 
+    "javax.", 
+    "org.", 
+    "mozilla."
+  };
+
+  private static final String[] SHARED_CLASSES = {
+    // Expose the GAE API classes
+    "com.google.appengine.api.LifecycleManager",
+    "com.google.apphosting.api.ApiProxy",
+    "com.google.apphosting.api.ApiStats",
+    "com.google.apphosting.api.CloudTrace",
+    "com.google.apphosting.api.CloudTraceContext",
+    "com.google.apphosting.api.DeadlineExceededException",
+    "com.google.apphosting.runtime.SessionData",
+    "com.google.apphosting.runtime.UncatchableError",
+
+    // Expose the standard APIs that are provided by the container
+    "javax.servlet.",
+    "javax.el.",
+    "javax.mail.", // TODO Review
+    
+    // Expose the standard APIs that are provided by the JVM
+    "com.oracle.",
+    "com.sun.",
+    "javax.accessibility.",
+    "javax.activation.",
+    "javax.activity.",
+    "javax.annotation.",
+    "javax.imageio.",
+    "javax.jws.",
+    "javax.lang.model.",
+    "javax.management.",
+    "javax.naming.",
+    "javax.net.",
+    "javax.print.",
+    "javax.rmi.",
+    "javax.script.",
+    "javax.security.",
+    "javax.smartcardio.",
+    "javax.sound.",
+    "javax.sql.",
+    "javax.swing.",
+    "javax.tools.",
+    "javax.transaction.",
+    "javax.xml.",
+    "jdk.",
+    "org.ietf.jgss.",
+    "org.jcp.xml.dsig.internal.",
+    "org.omg.",
+    "org.w3c.dom.",
+    "org.xml.",
+    "sun.",
+
+    // Expose classes needed for JSP and JSTL
+    "org.apache.jasper.runtime.",
+    "org.apache.jasper.JasperException",
+    "org.apache.el.ExpressionFactoryImpl",
+    "org.apache.tomcat.InstanceManager",
+    "org.apache.taglibs.",
+  };
+
+  @Override
+  public void preConfigure(WebAppContext context) {
+    for (String systemClass : SERVER_CLASSES) {
+      context.addServerClass(systemClass);
+    }
+    for (String gaeClass : SHARED_CLASSES) {
+      // Don't hide shared classes
+      context.prependServerClass("-" + gaeClass);
+      // Don't allow shared classes to be replaced by webapp
+      context.addSystemClass(gaeClass);
+    }
+  }
+
+  public void configure(WebAppContext context) throws Exception {
+    ClassLoader loader = context.getClassLoader();
+    try {
+      // Test if the appengine api is available
+      loader.loadClass(GAE_CHECK_CLASS);
+    } catch (Exception ex) {
+      VmRuntimeWebAppContext.logger.log(Level.FINE, "No appengined API", ex);
```

Comment:
Nit, would that be nicer?

```java
if (VmRuntimeWebAppContext.logger.isLoggable(Level.FINE)) {
  VmRuntimeWebAppContext.logger.log(Level.WARNING, "No appengine API jar included in WEB-INF/lib! Please update your SDK!", ex);
} else {
  VmRuntimeWebAppContext.logger.log(Level.WARNING, "No appengine API jar included in WEB-INF/lib! Please update your SDK!");
}
```

Regardless, I am not sure how useful the exception is going to be in any case.
And if you prefer it the way it is then `s/appengined/appengine/` and `s/including/included/`
```